pkg/alertmanager/amcfg_test.go

```diff
@@ -2885,7 +2885,110 @@ func TestTimeInterval(t *testing.T) {
 		})
 	}
 }
+func TestSanitizePushoverReceiverConfig(t *testing.T) {
+	logger := log.NewNopLogger()
 
+	for _, tc := range []struct {
+		name           string
+		againstVersion semver.Version
+		in             *alertmanagerConfig
+		expect         alertmanagerConfig
+		expectErr      bool
+	}{
+		{
+			name:           "Test pushover user_key_file/token_file takes precedence in pushover config",
+			againstVersion: semver.Version{Major: 0, Minor: 26},
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PushoverConfigs: []*pushoverConfig{
+							{
+								UserKey:     "foo",
+								UserKeyFile: "/path/use_key_file",
+								Token:       "bar",
+								TokenFile:   "/path/token_file",
+							},
+						},
+					},
+				},
+			},
+			expect: alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PushoverConfigs: []*pushoverConfig{
+							{
+								UserKey:     "foo",
+								UserKeyFile: "",
+								Token:       "bar",
+								TokenFile:   "",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Test pushover userkey/user_key_file one of must be configured",
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PushoverConfigs: []*pushoverConfig{
+							{
+								UserKey:     "foo",
+								UserKeyFile: "/path/use_key_file",
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name:           "Test pushover token/token_file one of must be configured",
+			againstVersion: semver.Version{Major: 0, Minor: 25},
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PushoverConfigs: []*pushoverConfig{
+							{
+								Token:     "bar",
+								TokenFile: "/path/token_file",
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name:           "Test pushover user_key_file/token_file dropped in pushover config for unsupported versions",
+			againstVersion: semver.Version{Major: 0, Minor: 25},
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PushoverConfigs: []*pushoverConfig{
+							{
+								UserKeyFile: "/path/use_key_file",
+								TokenFile:   "/path/token_file",
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
```

Comment:
Could we split this test in 2? We probably want this combinations

* version 0.25, UserKeyFile+token, we expect err
* version 0.25, UserKey+TokenFile, we expect err

Reply:
Done. Add unittest case combinations